core/dbt/contracts/graph/semantic_manifest.py

```diff
@@ -105,6 +106,12 @@ def _get_pydantic_semantic_manifest(self) -> PydanticSemanticManifest:
                     name=time_spine.standard_granularity_column,
                     time_granularity=standard_granularity_column.granularity,
                 ),
+                custom_granularities=[
+                    PydanticTimeSpineCustomGranularityColumn(
+                        name=custom_granularity.name, column_name=custom_granularity.column_name
+                    )
+                    for custom_granularity in time_spine.custom_granularities
+                ],
```

Comment:
The more we do this, the more I wish we had a better way to automatically translate them. Alas unless we're literally using the same object (or subclassing) we'll need the translation logic somewhere